.github/workflows/publish-testpypi.yml

```diff
@@ -21,29 +21,55 @@ jobs:
       - name: Install Python dependencies
         run: |
           python -m pip install --upgrade pip
-          make install requirements
+          make install
```

Comment:
This will install development requirements, why do we need them here?

Reply:
Good question. I'm not convinced they were necessary either. I've reduced the install requirements in https://github.com/unitaryfund/mitiq/pull/2320/commits/97e309f284e59445af330082b5d33c78ebb1c3fb.